pycbc/waveform/supernovae.py

```diff
@@ -0,0 +1,51 @@
+"""Generate core-collapse supernovae waveform for core bounce and
+subsequent postbounce oscillations.
+"""
+
+import numpy
+import h5py
+from pycbc.types import TimeSeries, FrequencySeries
+
+_principal_components = {}
+
+def get_td_corecollapse_bounce_signal(template=None, **kwargs):
+    """ Generates core bounce and postbounce waveform by using principal
+    component basis vectors from a .hdf file. The waveform parameters are the
+    coefficients of the principal components and the distance. The number of
+    principal components used can also be varied.
+    """
+
+    try:
+        principal_components = _principal_components['principal_components']
+    except KeyError:
+        pc_file = h5py.File(kwargs['principal_components_file'], 'r')
+        principal_components = pc_file['principal_components']
+        _principal_components['principal_components'] = principal_components
```

Comment:
Use a `with` here instead, as it will protect against dangling file handlers; i.e.:
```
with h5py.File(kwargs['principal_components_file'], 'r') as pc_file:
    principal_components = pc_file['principal_components']
    _principal_components['principal_components'] = principal_components
```